x/tokenfactory/keeper/admins_test.go

```diff
@@ -1,74 +1,270 @@
 
 	// Try setting admin to empty
-	_, err = msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), types.NewMsgChangeAdmin(suite.TestAccs[1].String(), denom, ""))
+	_, err = suite.msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), types.NewMsgChangeAdmin(suite.TestAccs[1].String(), suite.defaultDenom, ""))
 	suite.Require().NoError(err)
 	queryRes, err = suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
-		Denom: res.GetNewTokenDenom(),
+		Denom: suite.defaultDenom,
 	})
 	suite.Require().NoError(err)
 	suite.Require().Equal("", queryRes.AuthorityMetadata.Admin)
 }
+
+func (suite *KeeperTestSuite) TestMintDenom() {
+	var (
+		addr0bal = int64(0)
+	)
+
+	// Create a denom
+	suite.CreateDefaultDenom()
+
+	for _, tc := range []struct {
+		desc      string
+		amount    int64
+		mintDenom string
+		admin     string
+		valid     bool
+	}{
+		{
+			desc:      "denom does not exist",
+			amount:    10,
+			mintDenom: "factory/osmo1t7egva48prqmzl59x5ngv4zx0dtrwewc9m7z44/evmos",
+			admin:     suite.TestAccs[0].String(),
+			valid:     false,
+		},
+		{
+			desc:      "mint is not by the admin",
+			amount:    10,
+			mintDenom: suite.defaultDenom,
+			admin:     suite.TestAccs[1].String(),
+			valid:     false,
+		},
+		{
+			desc:      "success case",
+			amount:    10,
+			mintDenom: suite.defaultDenom,
+			admin:     suite.TestAccs[0].String(),
+			valid:     true,
+		},
+	} {
+		suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
+			// Test minting to admins own account
+			_, err := suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(tc.admin, sdk.NewInt64Coin(tc.mintDenom, 10)))
+			if tc.valid {
+				addr0bal += 10
+				suite.Require().NoError(err)
+				suite.Require().Equal(suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], suite.defaultDenom).Amount.Int64(), addr0bal, suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], suite.defaultDenom))
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestBurnDenom() {
+	var (
+		addr0bal = int64(0)
+	)
+
+	// Create a denom and mint
+	suite.CreateDefaultDenom()
+
+	// Make sure that the admin is set correctly
+	queryRes, err := suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
+		Denom: suite.defaultDenom,
+	})
+	suite.Require().NoError(err)
+	suite.Require().Equal(suite.TestAccs[0].String(), queryRes.AuthorityMetadata.Admin)
+
+	_, err = suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(suite.defaultDenom, 10)))
+	addr0bal += 10
+	suite.Require().NoError(err)
+	suite.Require().True(suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], suite.defaultDenom).Amount.Int64() == addr0bal, suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], suite.defaultDenom))
```

Comment:
What is this for? Making some tokens to burn?

These lines are hard to follow / the last seems non-standard equality checking syntax)